brion/plugin/compartmentReportHDF5.cpp

```diff
@@ -270,12 +265,31 @@ floatsPtr CompartmentReportHDF5::loadFrame(const float timestamp) const
         const hsize_t targetOffsets[2] = {0, firstCompartmentOffset};
         targetSpace.selectHyperslab(H5S_SELECT_SET, readCounts, targetOffsets);
 
-        dataset.read(buffer->data(), H5::PredType::NATIVE_FLOAT, targetSpace,
-                     space);
+        dataset.read(buffer, H5::PredType::NATIVE_FLOAT, targetSpace, space);
 
         firstCompartmentOffset += sourceSizes[1];
     }
+    return true;
+}
+
+Frames CompartmentReportHDF5::loadFrames(const float start,
+                                         const float end) const
+{
+    if (_offsets.empty())
+        return Frames();
+    lunchbox::ScopedWrite mutex(detail::_hdf5Lock);
+    return CompartmentReportCommon::loadFrames(start, end);
```

Comment:
What's wrong with keeping the hdf5 lock inside _loadFrame? As a matter of fact I'm quite sure you should keep it there. H5 is a crappy when it comes to multithreading. You have to options, either you compile it with "multithreading" support, which means every operations tries to lock a single global mutex, or you do it yourself.
If loadFrames could process multiple frames in parallel, this code would crash because every thread already holds the (external) lock and all will try to use HDF5 at the same time, which is a recipe for disaster (don't ask me why, but even when accessing different files, there are global data structures used by all of them).